File: DoItTest/ClientApp/src/domain/students/student.ts
export class Student {
    public constructor(
        public readonly id: string,
        public readonly firstName: string,
        public readonly lastName: string,
        public readonly patronymic: string | null,
        public readonly group: string
    ) { }

    public get fullName(): string {
        return this.lastName + " " + this.firstName + " " + (this.patronymic ?? "")
    }
}

export function mapToStudent(value: any): Student {
    return new Student(value.id, value.firstName, value.lastName, value.patronymic, value.group)
}

export function mapToStudents(value: any[]): Student[] {
    return value.map(mapToStudent);
}

File: DoItTest.Domain/Students/Student.cs
namespace DoItTest.Domain.Students
{
    public class Student
    {
        public Guid Id { get; }
        public String FirstName { get; }
        public String LastName { get; }
        public String? Patronymic { get; }
        public String Group { get; }

        public Student(Guid id, String firstName, String lastName, String? patronymic, String group)
        {
            Id = id;
            FirstName = firstName;
            LastName = lastName;
            Patronymic = patronymic;
            Group = group;
        }
    }
}
